MiniBlog/Controllers/UserController.cs

```diff
@@ -1,64 +1,50 @@
+using Microsoft.AspNetCore.Identity;
 using MiniBlog.Model;
 using MiniBlog.Stores;
 using Microsoft.AspNetCore.Mvc;
+using MiniBlog.Service;
 
 namespace MiniBlog.Controllers
 {
     [ApiController]
     [Route("[controller]")]
     public class UserController : ControllerBase
     {
-        [HttpPost]
-        public User Register(User user)
+        private IUserService userService;
+
+        public UserController(IUserService userService)
         {
-            if (!UserStoreWillReplaceInFuture.Instance.GetAll().Exists(_ => user.Name.ToLower() == _.Name.ToLower()))
-            {
-                UserStoreWillReplaceInFuture.Instance.Save(user);
-            }
+            this.userService = userService;
+        }
 
-            return user;
+        [HttpPost]
+        public ActionResult<User> Register(User user)
+        {
+            return Created(" ", userService.Register(user));
```

Comment:
"" in Created() shows how to get the resource for return user.  So, you may not return " " if you can.